### PR TITLE
Feature/standby hours

### PR DESCRIPTION
**Pull request by vladPinteaFrontmen**
# Changes
Employee can have standby hours toggled on and off
Timesheet now features standby hours if feature is turned on
Timesheets admin view shows stand by hours for all employees, with a label if it is no longer available. Travel expenses also always on (given that toggling it off would hide the history from when it was available). 
Employee cannot see the history of that when it isn't enabled anymore. 

additionally firestore collections moved to an enum

## Related issues
https://github.com/FrontMen/fm-hours/issues/171

## Added

New firestore collection for standby hours
Standby toggler in manage employee view
Standby hours in timesheet both on home and /timesheets

## Changed
Time records service - methods made generic to be able to perform CRUD on both time reports and standby hours

## How to test
1. Login as admin
2. Go to employees and toggle on standby hours
3. Go to home page and see a new row to fill. 
4. Fill it, save it, see it persists. You can also check firestore that it correctly updated
5. Go to /timesheets and check the sheet you filled in - see the new standby area
6. Go to another employee's timesheets, who doesn't have stand by enabled, see a label warning you it isn't available
7. Do the same for an employee without travel enabled

## Screenshots
image
image
image

after rebase with Dean's changes 
image

Lasse's enhancement recommandation
image

### CODE REVIEW

**Comment by lassecapel** [comment]
can we use a badge instead of the text "not active for employee" and maybe make the whole table grey instead of primary color

File: store/records/actions.ts [hunk 1]
```diff
@@ -137,11 +143,21 @@ const actions: ActionTree<RecordsStoreState, RootStoreState> = {
...
+
+    const timeRecords = await this.app.$timeRecordsService.saveEmployeeRecords<TimeRecord>({
       employeeId: payload.employeeId,
       timeRecords: timeRecordsToSave,
     });
 
+    const standByRecords = await this.app.$timeRecordsService.saveEmployeeRecords<StandbyRecord>({
+      employeeId: payload.employeeId,
+      timeRecords: standByRecordsToSave,
+    }, "standby_records");
```
Comment (italoteix) [inline store/records/actions.ts:159]: Can we use the ENUM here?

**Review by italoteix — CHANGES_REQUESTED** [review]
While testing I couldn't create new timesheets (the project modal didn't show up).
Also, for some employees, the Standby table is still editable even if the standby hours aren't enable for him

**Comment by vladPinteaFrontmen** [comment]
@italoteix updated: removed the total from standby; adding a project reenabled; 
admins should still be able to edit values even for fields that are no longer enabled.

**Comment by italoteix** [comment]
Tested again. It seems that the employees that don't have the `standBy` field at the database (the ones that weren't updated using this branch) are showing up the Stand-by table as it was true. How to reproduce it: 
- Delete the `standBy` field of a user directly on the database or go to the timesheets of some user that doesn't have it (like Dean).
- Check that the stand by fields are enabled to him (should default to not)

**Comment by italoteix** [comment]
If for some reason the employee logs hours only on stand-by fields, the app will show the "empty timesheet" state after saving. How to reproduce it: 
- Go to an empty week for of a employee that has the standBy enabled
- Add a new timesheet
- Log hours only in the `Stand-by hours` table
- Save it

**Review by italoteix — APPROVED** [review]

**Pull request merged**